src/main/java/graphs/GraphSearch.java

```diff
@@ -158,4 +159,102 @@
         }
         return nodeList;
     }
+
+
+    /*
+        djikstras algo:
+        takes: graph, source, target
+
+        map of V to distance from source
+            - instantiate at MAX_INT
+        map of V's previous node
+
+        set of V
+
+        // set source distance to zero
+        distanceMap.put(source, 0)
+        while(! setOfV.isEmpty())
+
+            Node currentNode = setOfV.pop()
+            update distance values for all nodes from nextNode
+            ie get incident edges and
+            update distance for each node = min(distance(node), distance(currentNode) + edge(currentNode, nextNode).getWeight())
+                if distance < current distance + edge weight
+                    update distance
+                    update previous node
+
+            // sort algo that takes list of nodes and map to their current distances and sorts nodes
+            find next node that is ! in settled and shortest distance (ie distance(currentNode) + edge(currentNode, nextNode).getWeight())
+            shortest distance nextNode gets added to settled
+
+        eventually every node will have a non infinite distance from source
+
+
+     */
+
+    public static List<Node> djikstraShortestPath(Graph graph, Node start, Node finish) {
+        if (graph == null || graph.isEmpty()) {
+            throw new IllegalArgumentException("empty or null graph");
+        }
+        if(start == null || finish == null) {
+            throw new IllegalArgumentException("null start node or target node");
+        }
+        if(! (graph.getNodes().contains(start) && graph.getNodes().contains(finish))) {
+            return new LinkedList();
```

Comment:
it seems to me like this would be exceptional. the caller provided a graph that doesn't contain the start or finish node. empty list indicates that there is no path between start and finish, but implicitly they are both contained in the graph. my vote is a throw here (though if you feel strongly about this, i think its alright)